torch/csrc/jit/codegen/cuda/tensor_view.cpp

```diff
@@ -344,6 +180,117 @@ TensorView* TensorView::computeAt(TensorView* consumer, int axis) {
   return this;
 }
 
+TensorView* TensorView::split(int axis, int factor) {
+  if (axis < 0)
+    axis += domain()->nDims();
+
+  TORCH_CHECK(
+      axis >= 0 && axis < domain()->nDims(),
+      "Trying to split axis outside of TensorView's range.");
+
+  if (getComputeAtView() != nullptr)
+    if (axis < getComputeAtAxis())
+      TORCH_CHECK(false, "Cannot split axis within compute at range.");
+
+  setDomain(domain()->split(axis, factor));
+  return this;
+}
+
+// Merge "axis" and "axis+1" into 1 dimension
+TensorView* TensorView::merge(int axis) {
+  if (axis < 0)
+    axis += domain()->nDims();
+
+  TORCH_CHECK(
+      axis >= 0 && axis + 1 < domain()->nDims(),
+      "Trying to merge axis outside of TensorView's range.");
+
+  if (getComputeAtView() != nullptr)
+    if (axis + 1 < getComputeAtAxis())
+      TORCH_CHECK(false, "Cannot merge axis within compute at range.");
+
+  setDomain(domain()->merge(axis));
+  return this;
+}
+
+// Reorder axes according to map[old_pos] = new_pos
+TensorView* TensorView::reorder(const std::unordered_map<int, int>& axis2pos_) {
+  // START VALIDATION CHECKS
```

Comment:
What's the point of duplicating the checks here, as we also do it in TensorDomain::reorder?